micro/service.go

```diff
@@ -482,6 +497,9 @@ func (c *Config) valid() error {
 	if !semVerRegexp.MatchString(c.Version) {
 		return fmt.Errorf("%w: version: version should not be empty should match the SemVer format", ErrConfigValidation)
 	}
+	if _, err := url.Parse(c.APIURL); err != nil {
```

Comment:
I know we call it as api URL, but here we are making it so that it must be - imho this value should just be passthrough.

Reply:
I think if it's `api_url`, we should validate... `url.Parse` in go a rather loose validation, allowing no schema, relative urls etc., so it should give some flexibility but disallow clearly invalid strings (newlines, urls starting with `:` etc.)